### PR TITLE
#972: POC for new DatePicker (closes #972)

### LINKED ISSUE #972: POC for new DatePicker
## requirements
Since `rc-datepicker` is missing important features and maintaining it is becoming more and more time consuming, we'd like to implement a new `DatePicker` wrapping a more famous one, like `react-dates`

## specs
- This POC is an attempt to maintain the same old API wrapping a component like `react-dates`.
- Month/Year fast navigation is not strictly required. Picking dates in the past/future (like birth dates) will be covered by a different, more appropriate component.

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by veej**
Closes #972

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,6 +48,8 @@
     "linkify-it": "^2.0.3",
     "lodash": "^4.6.1",
     "mobile-detect": "^1.2.1",
+    "react-addons-shallow-compare": "^15.6.0",
```
Comment (FrancescoCioria) [inline package.json]: is this really used?

File: src/DatePicker/DatePicker.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import cx from 'classnames';
+import { SingleDatePicker } from 'react-dates';
+import FlexView from 'react-flexview';
+import { props, t, skinnable } from '../utils';
+import moment from 'moment';
```
Comment (FrancescoCioria) [inline src/DatePicker/DatePicker.js]: if we need `moment` we should add it to the deps

File: src/DatePicker/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# NewDatePicker
```
Comment (FrancescoCioria) [inline src/DatePicker/README.md]: I think this is outdated, try running `npm run generate-all-readmes`

File: src/DatePicker/examples/default.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+class Example extends React.Component {
+
+  constructor() {
```
Comment (FrancescoCioria) [inline src/DatePicker/examples/default.js]: minor: replace `constructor` with cleaner `state = {}`

**Comment by FrancescoCioria** [comment]
@veej also could you add the `d.ts` typings?

File: package.json (hunk 1 above)
Comment (veej) [inline package.json]: Unfortunately yes, it's needed by [react-dates](https://github.com/airbnb/react-dates) (see `Install dependencies` section)

**Pull request merged**